gnovm/pkg/gnolang/machine.go

```diff
@@ -2231,6 +2267,27 @@ func (m *Machine) String() string {
 	return builder.String()
 }
 
+func (m *Machine) ExceptionsStacktrace() string {
+	var builder strings.Builder
+
+	i := 0
+	for i < len(m.Exceptions) {
+		ex := m.Exceptions[i]
+		builder.WriteString(fmt.Sprintf("panic %s\n", ex.Sprint(m)))
+		builder.WriteString(fmt.Sprintf("%s", ex.Stacktrace.String()))
+
+		if i == 0 && len(m.Exceptions) > 2 {
+			builder.WriteString(fmt.Sprintf("... %d panic(s) elided ...\n", len(m.Exceptions)-2))
+			i = len(m.Exceptions) - 1
+			continue
+		}
+
+		i++
+	}
+
+	return builder.String()
+}
+
```

Comment:
In an effort to avoid repeating three lines of code, this function is unclear.

```suggestion
func (m *Machine) ExceptionsStacktrace() string {
	if len(m.Exceptions) == 0 {
		return ""
	}

	var builder strings.Builder

	ex := m.Exceptions[0]
	builder.WriteString("panic: " + ex.Sprint(m) + "\n")
	builder.WriteString(ex.Stacktrace.String())

	switch {
	case len(m.Exceptions) > 2:
		fmt.Fprintf(&builder, "... %d panic(s) elided ...\n", len(m.Exceptions)-2)
		fallthrough // to print last exception
	case len(m.Exceptions) == 2:
		ex = m.Exceptions[len(m.Exceptions)-1]
		builder.WriteString("panic: " + ex.Sprint(m) + "\n")
		builder.WriteString(ex.Stacktrace.String())
	}

	return builder.String()
}
```